.github/workflows/ci.yml

```diff
@@ -0,0 +1,70 @@
+# This workflow will do a clean install of the dependencies and run tests across different versions
+#
+# Replace <track> with the track name
+# Replace <image-name> with an image to run the jobs on
+# Replace <action to setup tooling> with a github action to setup tooling on the image
+# Replace <install dependencies> with a cli command to install the dependencies
+# Replace <code-extensions> with file extensions that should trigger the workflow
+#
+# Find Github Actions to setup tooling here:
+# - https://github.com/actions/?q=setup&type=&language=
+# - https://github.com/actions/starter-workflows/tree/main/ci
+# - https://github.com/marketplace?type=actions&query=setup
+#
+# Requires scripts:
+# - bin/run_lints.R
+# - bin/run_tests.R
+
+name: r / main
+
+on:
+  push:
+    branches: [master, main]
+  pull_request:
+    branches: [master, main]
+  workflow_dispatch:
+
+jobs:
+  precheck:
+    runs-on: macOS-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      
+      - uses: r-lib/actions/setup-R@v1
+        with:
+          r-version: 3.6.3
+
+      - name: Install project dependencies
+        run: |
+          install.packages(c("testthat", "jsonlite", "lintr", "magrittr"))
+        shell: Rscript {0}
+
+      - name: Run exercism/r ci pre-check (checks config, lint code) for all exercises
+        run: |
+          source("bin/run_lints.R")
+        shell: Rscript {0}
```

Comment:
Was there a specific reason not to run the linting in the `ci` step below? If it was integrated there, it would be a bit faster due to not having to do the checkout, setup of R and install package steps being doubled here.

Reply:
Only did it this way cos I was working from the template suggested here: https://github.com/exercism/problem-specifications/blob/master/workflows/ci.yml

But I guess the steps were separated in the template cos linting usually only needs to be checked on a single stable version (and possibly has fewer dependencies) whereas one generally wants to run the tests on multiple versions (and more of the dependencies are required here) in case of any incompatibilities?

Is that tradeoff worth it or do you think I should rather just combine the lint and test checks in the CI step?

Reply:
Ah right. It probably makes sense to adhere to the default workflow then. I missed that, sorry.

Reply:
Cool no worries. Thanks so much for the review!